### DISCUSSION TITLE
The problem of using ReactiveUI for automatic settings saving in Avalonia MVVM pattern.

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#14271] [Q&A] [answered]
My program automatically saves various settings on the window, and it works by saving all the content on the window after triggering a PropertyChanged event. However, I found that when I load the saved content from a file, every change in each control will trigger the PropertyChanged event once, causing the program to save every time a control is changed. This causes the program to hang. Is there a better implementation method?

**Comment by timunie** [comment, marked as answer]
I only save user settings on App shutdown, but that may be too late for someone else. However doing that on any property sounds expensive. Consider to throttle it down a bit by using a DispatcherTimer for example. Also make a flag _isLoading which you can use to ignore these changes during load.

**Comment by thevortexcloud** [comment]
You can just disable change notifications. ReactiveUI has a helper method for that returns an `IDisposable` so you can wrap it in a using. I think it's something like `SuspendChangeNotifcations`. I have had to use it in the past to get any kind of reasonable performance in an app that does *a lot* of rapid data changes. I think I went from about ~40 seconds to ~1.2 seconds by doing that.

I personally though normally just save on explicit confirmation (EG clicking a button) or the window being deactivated. Which happens when it's closed.
